libcontainer/apparmor/apparmor_linux.go

```diff
@@ -26,14 +26,19 @@ func isEnabled() bool {
 }
 
 func setProcAttr(attr, value string) error {
-	// Under AppArmor you can only change your own attr, so use /proc/self/
-	// instead of /proc/<tid>/ like libapparmor does
-	attrPath := "/proc/self/attr/apparmor/" + attr
-	if _, err := os.Stat(attrPath); errors.Is(err, os.ErrNotExist) {
+	attr = utils.CleanPath(attr)
+	attrSubPath := "attr/apparmor/" + attr
+	if _, err := os.Stat("/proc/self/" + attrSubPath); errors.Is(err, os.ErrNotExist) {
 		// fall back to the old convention
-		attrPath = "/proc/self/attr/" + attr
+		attrSubPath = "attr/" + attr
 	}
 
+	// Under AppArmor you can only change your own attr, so there's no reason
+	// to not use /proc/thread-self/ (instead of /proc/<tid>/, like libapparmor
+	// does).
```

Comment:
I'm not sure what you mean by "do the opposite"? We still don't use `/proc/<tid>`.

(To be honest, I'm not sure this behaviour even needs a comment because it is obviously correct, but I've kept it from when we switched away from libapparmor several years ago.)

Reply:
SGTM.

I meant the comment before says: don't use proc/<tid>
Now it says: there is no reason to not use proc/thread-self/.

I think yesterday I was having issues with a "logic not", this SGTM :)